exporters/prometheus/src/exporter_utils.cc

```diff
@@ -258,16 +258,37 @@ void PrometheusExporterUtils::SetData(std::vector<T> values,
 void PrometheusExporterUtils::SetMetricBasic(prometheus_client::ClientMetric &metric,
                                              const metric_sdk::PointAttributes &labels)
 {
-  // auto label_pairs = ParseLabel(labels);
-  if (!labels.empty())
+  if (labels.empty())
   {
-    metric.label.resize(labels.size());
-    size_t i = 0;
-    for (auto const &label : labels)
+    return;
+  }
+
+  // Concatenate values for keys that collide after sanitation.
+  // Note that attribute keys are sorted, but sanitized keys can be out-of-order.
```

Comment:
The SDK doesn't sort the attribute keys. Unless the instrumentation library is recording them in sorted order, I am not sure if below logic would work.

Reply:
The type is `OrderedAttributeMap` which is an alias for `std::vector<std::string, ...>`. I see that a similar unordered type exists, but that's not the one used here.